mixer/cmd/versions.go

```diff
@@ -0,0 +1,141 @@
+// Copyright © 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+
+	"github.com/clearlinux/mixer-tools/builder"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var updateVersionsCmd = &cobra.Command{
+	Use:   "update-versions",
+	Short: "Update the mix and upstream versions",
+	Long: `Choose the next mix and upstream versions based.
+
+Without any flags, it increments mix by 10 and upstream to the latest
+version of the same format used by current upstream. Use --mix-version
+and --upstream-version to set a specific version. By default the
+versions are shown but not set, to set use --set (-s).
+
+This command is aware of format bumps and will not automatically
+propose a version change cross different formats in upstream. At the
+moment format bumps must be handled manually when using Mixer.
+`,
+	Run: runUpdateVersions,
+}
+
+var updateVersionsFlags struct {
+	mixVersion      uint32
+	upstreamVersion uint32
+	set             bool
+}
+
+func init() {
+	RootCmd.AddCommand(updateVersionsCmd)
+	updateVersionsCmd.Flags().StringVarP(&config, "config", "c", "", "Builder config to use")
+	updateVersionsCmd.Flags().Uint32Var(&updateVersionsFlags.mixVersion, "mix-version", 0, "Next mix version (0 means increment current one)")
```

Comment:
Would prefer "increment current value by 10"; if I were a newcomer that hadn't paid attention to CLR's versioning, I would interpret "increment" as `+= 1`

Reply:
Fixed the help to reflect that (and explain why 10), also giving the user an --increment flag.